Automattic/wp-calypso
# Signup: Add email suggestion to social first passwordless email sign up

Related to https://github.com/Automattic/wp-calypso/issues/53380

## Proposed Changes

* These changes provide users with a hint to correct their email address in case of easy-to-catch typos.
`agru@gmmail.com` -> `agru@gmail.com`

<img width="440" alt="CleanShot 2023-11-10 at 09 29 32@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/7df7eec5-39cf-43ad-acc5-d780aef10cea">

We are currently only targeting these 10 email providers based on their data from https://www.gmass.co/domains:

```
const validDomains = [
		'gmail.com',
		'yahoo.com',
		'hotmail.com',
		'aol.com',
		'outlook.com',
		'comcast.net',
		'icloud.com',
		'msn.com',
		'hotmail.co.uk',
		'sbcglobal.net',
	];

```

## Testing Instructions



* Checkout branch or use live link
* Navigate to `/start/user` (logged out)
* Enter an email with an invalid domain (such as gmail.com, hotmail.com, etc.)
* Verify the message is suggesting a domain that makes sense for the typo in your email
* Click on the underlined suggested domain to update the input field, then click "Continue" and verify the account is successfully created.
* Enter a random email, trigger the domain suggestion, ignore the domain suggestion message, click "Continue" and verify the account is successfully created.


Fixes https://github.com/Automattic/wp-calypso/issues/53380

## Pre-merge Checklist
